core/block_validator.go

```diff
@@ -133,18 +133,18 @@ func (v *BlockValidator) ValidateState(block *types.Block, statedb *state.StateD
 		return fmt.Errorf("invalid merkle root (remote: %x local: %x) dberr: %w", header.Root, root, statedb.Error())
 	}
 	if blockEw := block.ExecutionWitness(); blockEw != nil {
-		parent := v.bc.GetHeaderByNumber(header.Number.Uint64() - 1)
+		parent := v.bc.GetBlockByHash(header.ParentHash)
```

Comment:
We can't rely on canonical chain until the node is synced, so in general this is a safer option for all use cases (syncing, tests, block creation).